tests/test_basic_functionality.py

```diff
@@ -458,6 +458,30 @@ def test_to_dict_yaml(test_spec: Path) -> None:
     assert yaml_config == lct.to_dict()
 
 
+def test_getter() -> None:
```

Comment:
It is a good rule of thumb to try to have as few asserts as possible in a single test. You have 9 asserts in this test, so if the first one fails, the remaining checks never get run. If I made a change that broke all of these cases, I wouldn't know anything beyond the first failure, so I might fix the first, and then have it fail on the next assertion, on and on until the 9th one. 
If the asserts were all broken into separate tests or different parameterized versions of the same test, I would be able to see that all 9 tests failed the first time it ran, so I could fix them all at once.

Reply:
Also, as long as the tests are named usefully, it makes it easier to see what cases our tests are covering.

Reply:
@hussain-jafari and this is going to be _super_ important in the psp test suite since that takes soooo long to run!